backend/src/org/commcare/session/CommCareSession.java

```diff
@@ -604,6 +595,25 @@ public boolean executeStackOperations(Vector<StackOperation> ops, EvaluationCont
         return false;
     }
 
+    private Boolean performPush(StackOperation op, SessionFrame matchingFrame,
+                                boolean isNewFrame, boolean currentFramePushed,
+                                SessionFrame onDeck, EvaluationContext ec) {
+        if (op.isOperationTriggered(ec)) {
+            //If we don't have a frame yet, this push is targeting the
+            //frame on deck
+            if (matchingFrame == null) {
+                matchingFrame = onDeck;
+            }
+
+            for (StackFrameStep step : op.getStackFrameSteps()) {
+                matchingFrame.pushStep(step.defineStep(ec));
+            }
+
+            return pushNewFrame(matchingFrame, isNewFrame, currentFramePushed);
+        }
+        return null;
```

Comment:
why not stick to your original strategy of returning `currentFramePushed` in this case, so that you don't have to do the null check after calling this method?